### ISSUE TITLE
Apps fail to initialise (after update to 4.0.0)

### ISSUE THREAD

**Issue by patrick984** [home-assistant/appdaemon#845]
After upgrading to version 4.0.0, apps are no longer initialising. 
I have applied the config changes described in [Upgrading from 3.x](https://appdaemon.readthedocs.io/en/4.0.0/UPGRADE_FROM_3.x.html).

appdaemon.log
```log
2020-01-16 19:57:38.834310 INFO AppDaemon: Loading App Module: /home/homeassistant/.homeassistant/apps/hello.py
2020-01-16 19:57:38.842098 INFO AppDaemon: Initializing app hello_world using class HelloWorld from module hello
2020-01-16 19:57:38.869434 WARNING AppDaemon: Logged an error to /home/homeassistant/.homeassistant/appdaemon.err.log
2020-01-16 19:57:38.904555 WARNING AppDaemon: Unable to find module hello_world - initialize() skipped
2020-01-16 19:57:38.912906 INFO AppDaemon: App initialization complete
```
appdaemon.err.log
```log
2020-01-16 19:57:38.847204 WARNING hello_world: ------------------------------------------------------------
2020-01-16 19:57:38.851669 WARNING hello_world: Unexpected error initializing app: hello_world:
2020-01-16 19:57:38.853852 WARNING hello_world: ------------------------------------------------------------
2020-01-16 19:57:38.866824 WARNING hello_world: Traceback (most recent call last):
  File "/srv/appdaemon/lib/python3.7/site-packages/appdaemon/app_management.py", line 820, in check_app_updates
    await self.init_object(app)
  File "/srv/appdaemon/lib/python3.7/site-packages/appdaemon/app_management.py", line 280, in init_object
    self.AD, name, self.AD.logging, app_args, self.AD.config, self.app_config, self.AD.global_vars
  File "/srv/appdaemon/lib/python3.7/site-packages/appdaemon/plugins/hass/hassapi.py", line 46, in __init__
    adbase.ADBase.__init__(self, ad, name, logging, args, config, app_config, global_vars)
  File "/srv/appdaemon/lib/python3.7/site-packages/appdaemon/adbase.py", line 75, in __init__
    self.dashboard_dir = self.AD.http.dashboard_dir
AttributeError: 'NoneType' object has no attribute 'dashboard_dir'

2020-01-16 19:57:38.867885 WARNING hello_world: ------------------------------------------------------------
```

**Comment by Flurkmark** [comment]
Excatly the same here. Did a clean install in venv, only loading the HelloWorld example, same result.
python 3.7.5, ubuntu  
For fun, caught the first exception only to get a new one:
```
class Hello(hass.Hass):
 
  def __init__(self, *args, **kwargs):
    try:
      super(Hello, self).__init__(*args, **kwargs)
    except AttributeError:
      pass
  def initialize(self):
    self.log("Hello from AppDaemon")
    self.log("You are now ready to run Apps!")
```
Result:
``` File "/srv/appdaemon/lib/python3.7/site-packages/appdaemon/app_management.py", line 145, in initialize_app
    await utils.run_in_executor(self, init)
  File "/srv/appdaemon/lib/python3.7/site-packages/appdaemon/utils.py", line 276, in run_in_executor
    response = future.result()
  File "/usr/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/peter/appdaemon/apps/hello.py", line 17, in initialize
    self.log("Hello from AppDaemon")
  File "/srv/appdaemon/lib/python3.7/site-packages/appdaemon/adapi.py", line 141, in log
    logger = self.logger
AttributeError: 'Hello' object has no attribute 'logger'
```
Config:
```appdaemon:
  threads: 10
  plugins:
    HASS:
      type: hass
      ha_url: <ip>
      token: <token>
  logs:
    main_log:
      filename: <filename>
    access_log: 
      filename: <filename>
    error_log:
      filename: <filename>

  time_zone: '<tzA'
  latitude: <lat>
  longitude: <long>
  elevation: 0

**Comment by ReneTode** [comment]
because noone had that problem, when we were running in beta ;)

**Comment by Flurkmark** [comment]
Don't get testy. Let me rephrase:

Why am I hitting this bug, is there something missing in my config that would as easily fix it?
Since it's a release I would assume it would be in the docs.

**Comment by benleb** [comment]
Not testy, amused 😄👍

**Comment by Flurkmark** [comment]
@ReneTode Okay, thanks. So it's simply not that the fix in the PR breaks something else, but rather that most users have set up something that I have not. Not using api or dashboard and such. I take it that I can just apply the PR fix and go on.

**Comment by patrick984** [comment]
Can confirm installing #840 fixes the error. Thanks :)

And I learned pip can install PRs directly, pretty cool:
```
pip install git+https://github.com/home-assistant/appdaemon.git@refs/pull/840/merge
```

**Comment by ReneTode** [comment]
thx for sharing that @patrick984 

and for @patrick984 and @Flurkmark i would advise to consider making sure you have access to the admin interface. (which would also have worked as a fix)
the admin interface will give you great insight in your automations, active callbacks, the amount of times they are called, etc. which is extremely helpfull with debugging and with optimising your apps.

**Comment by Flurkmark** [comment]
@patrick984 Neat tip, thanks. 
@ReneTode Looking into it now that I have async working with turn_on.. thanks.

**Comment by ReneTode** [comment]
@Flurkmark i can say enough: only use async when you are aware of the risks, and when you are absolutely sure that you wont get any errors, when your apps are running.

**Comment by acockburn** [comment]
Fix is in the next release.

**Comment by johnruemker** [comment]
It appears this was never fixed.  

Is this change coming to appdaemon:latest - for those of us that aren't interested in enabling the admin interface, API, or dashboard?  The solution that was pushed in the comments  https://github.com/AppDaemon/appdaemon/pull/840 was to work-around the problem by enabling the http component.  I don't want to do that.  

Can we have the fix from that PR?  Using appdaemon:dev corrects the problem for me.

**Comment by ReneTode** [comment]
there was a fix, but its still incomplete in the version after it.
i guess the fix was after the latest release already, because else you would have had the same problem with dev ;) 

in general for AD app users its highly advised to use admin anyway. because its a great help for debugging

**Comment by johnruemker** [comment]
> there was a fix, but its still incomplete in the version after it.
> i guess the fix was after the latest release already, because else you would have had the same problem with dev ;)
> 

Where is that fix being tracked?  This issue was closed saying that it would be fixed, and you're saying the fix hasn't been released yet.  I'd like to follow the actual tracker for this so I know what's happening. 

> in general for AD app users its highly advised to use admin anyway. because its a great help for debugging

The admin console is an optional feature, isn't it?  Can we please have a proper fix then so I am not forced to use an optional feature I don't want?  

Thanks

**Comment by ReneTode** [comment]
it does work correct in dev right?
so its fixed, just not released.

by the way the error showed in the opening from this issue is also fixed and released in released versions from AD.

so please check which version from AD you tried that did have the problem.

(and please dont use "latest" in docker, because that has a problem also and doesnt use latest)

**Comment by johnruemker** [comment]
> it does work correct in dev right?
> so its fixed, just not released.
> 
> by the way the error showed in the opening from this issue is also fixed and released in released versions from AD.
> 
> so please check which version from AD you tried that did have the problem.
> 
> (and please dont use "latest" in docker, because that has a problem also and doesnt use latest)
> 


The fact that I was still seeing the error reported in this issue was due to "appdaemon:latest" tag not pointing to the latest release. 
 It is version 4.0.3 that is currently tagged as latest.  dev is 4.0.6.  Latest stable release according to github is 4.0.5.  Using 4.0.5 corrects the problem reported in this issue, so I am all set. 

I recommend this project fix its tags.  

> and you can also stay with dev.
> there hasnt much changed since the latest release so its just as stable.

**Comment by ReneTode** [comment]
at least you now know that its always possible that a tag is pointing wrong, and that it helps to check out if you got indeed the latest version ;) 

and the tag problem is known and will be corrected at next release.
